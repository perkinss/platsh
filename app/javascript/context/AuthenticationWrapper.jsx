import React, { useEffect, useState, useContext } from 'react';
import User from '../model/user'

export const RootAuthContext = React.createContext();
export default ({ children }) => {
  let anyAuthVal,anyAuthData;
  try {
    anyAuthVal=JSON.parse(window.sessionStorage.getItem('authenticated'));
  }
  catch(err) {
    anyAuthVal=false;
  }
  try {
    anyAuthData = JSON.parse(window.sessionStorage.getItem('authData'))
  }
  catch(err) {
    anyAuthData = null;
  }
  const guestUser = new User({})
  const prevAuth = (anyAuthVal ? anyAuthVal : false);
  const prevAuthData = (anyAuthData ? anyAuthData : null);
  const prevUser = (anyAuthData ? new User(anyAuthData) : guestUser);
  const [authenticated, setAuthenticated] = useState(prevAuth);
  const [authData, setAuthData] = useState(prevAuthData);
  const [currentUser, setCurrentUser] = useState(prevUser);
  useEffect(
    () => {
      window.sessionStorage.setItem('authenticated', JSON.stringify(authenticated));
      window.sessionStorage.setItem('authData', (authData ? JSON.stringify(authData):null));
      setCurrentUser(authData ? new User(authData) : guestUser)
    },
    [authenticated, authData]
  );
  const defaultContext = {
    authenticated, setAuthenticated,
    authData, setAuthData,
    currentUser, setCurrentUser
  };
  return (
    <RootAuthContext.Provider value={defaultContext}>
      {children}
    </RootAuthContext.Provider>
  );
};
