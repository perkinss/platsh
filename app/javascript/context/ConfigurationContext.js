import React from 'react'

export const configuration = {
    sections: [
        {
            id: null,
            courses: [],
            name: '',
            assessment: [
                {
                    type: 'test',
                    name: '',
                    tasks: [
                    {
                        name: '',
                        available_standards: [
                            {
                                id: -1,
                                level: ''
                            }
                        ],
                        competencies: [

                        ]
                    },
                    ]
                }
            ],
            enrollees: [
                1, 2, 3, 22
            ]
        },
    ],

}


const ConfigurationContext =  React.createContext(configuration);

export { ConfigurationContext }
