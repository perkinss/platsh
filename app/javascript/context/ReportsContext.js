import React from 'react'

export const reports = {
    students: [],
    sections: [],
    selected_section: null,
    selected_student: null,
    section_overview_marks: {},
    section_competency_scores: {},
    student_section_marks: {},
    heat_map_colors: []
}


const ReportsContext =  React.createContext(reports);

export { ReportsContext }
