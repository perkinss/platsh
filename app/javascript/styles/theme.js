import {createMuiTheme} from "@material-ui/core/styles";

const theme = createMuiTheme({
    palette: {
        primary: {
            main: '#512da8',
        },
        secondary: {
            main: '#ab47bc',
        },
        type: 'dark'
    },
    slider: {
        selectionColor: '#F76B1C',
        handleFillColor: '#F76B1C',
        thumb: 'secondary'
    }

})

export default theme
