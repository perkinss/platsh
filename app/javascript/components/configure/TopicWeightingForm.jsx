import React, { useContext, useState, useEffect } from 'react'
import {
    Typography, Paper, Grid, FormControl, InputLabel, Select, Input, MenuItem, Grow, TextField,
    Table, TableHead, TableRow, TableCell, TableBody, Button
} from '@material-ui/core'
import { useTheme,makeStyles } from "@material-ui/core/styles";
import { ConfigurationContext } from "../../context/ConfigurationContext";
import ReactApexChart from 'react-apexcharts'

const useStyles = makeStyles(theme => ({
    weights: {
        padding: theme.spacing(2,0)
    }
}));

export default function TopicWeightingForm(props) {
    let { classes, setShowSnackbar } = props;
    classes = { ...classes, ...useStyles() }
    const theme = useTheme()
    const {dispatch, state} = useContext(ConfigurationContext)

    const [weightings, setWeightings] = useState([])
    const [currentCourseWeighting, setCurrentCourseWeighting] = useState([])
    const [values, setValues] = useState([])

    const [labels, setLabels] = useState([])
    const [formHasUnsavedChanges, setFormHasUnsavedChanges] = useState(false)

    const courses = {}
    state.sections.forEach(section => {
        section.courses.forEach(course => {
            courses[course.id] = course.title
        })
    })

    useEffect(() => {
        fetch('/api/content_weighting/get_for_user_courses')
            .then(res => res.json())
            .then(
                (result) => {
                    setWeightings(result)
                },
                (error) => {
                    //TODO:Error snackbar
                    console.log("Error:", error)
                }
            )
    }, [])

    useEffect(() => {
        if (weightings && state.selected_course_id) {
            let course = weightings.find((weighting) => {
                return weighting.id === Number(state.selected_course_id)
            })
            let weights = [], names = []
            course.content_weights.forEach(content => {
                weights.push(content.weight)
                names.push(content.name)
            })
            setValues(weights)
            setLabels(names)
            setCurrentCourseWeighting(course.content_weights)
        }
    }, [weightings, state.selected_course_id])

    const handleChangeWeight = (event, w) => {
        w.weight = event.target.value.replace(/\D/g, '')
        if (event.target.value.length > 0 ) {
            w.weight = Number(w.weight)
        }
        let copy = Array.from(currentCourseWeighting)
        let index = copy.findIndex(a => a.id === w.id )
        copy[index] = w
        setCurrentCourseWeighting(copy)
        setFormHasUnsavedChanges(true)
    }

    const handleBlurWeight = (event, w) => {
        let wt = Array.from(currentCourseWeighting)
        let idx = wt.findIndex(a => a.id === w.id )
        if (event.target.value.length < 1) {
            w.weight = 1
            wt[idx] = w
            setCurrentCourseWeighting(wt)
        }
        let weights = Array.from(values)
        weights[idx] = Number(event.target.value)
        setValues(weights)
    }

    const handleSaveWeights = () => {
        const courseWeight = { course: state.selected_course_id, weights: currentCourseWeighting.map( (wt) => { return { content_id: wt.id, weight: wt.weight } }) }
        let token = document.querySelector('head>meta[name="csrf-token"]').content
        let body = JSON.stringify(courseWeight)
        fetch('/api/content_weighting/update',
            {
                method: 'POST',
                body: body,
                headers: {
                    'X-Requested-With': 'XMLHttpRequest',
                    'X-CSRF-Token': token,
                    'Content-Type': 'application/json',
                    'Accept': 'application/json'
                },
                credentials: 'same-origin'
            })
            .then(res => res.json())
            .then(response => {
                setShowSnackbar({open:true,variant:"success",message:"Successfully Saved!"})
            })
            .catch(error => console.log("ERROR: ", error))


        setFormHasUnsavedChanges(false)
    }

    const saveIsDisabled = () => {
        return !formHasUnsavedChanges
    }

    const chartOptions = {
        chart: {
            width: 300,
            type: 'pie',
            borderColor: theme.palette.grey[800],
            background: theme.palette.grey[800],
        },
        legend: {
            show: window.innerWidth > 800,
            verticalAlign: 'left',
            position: 'bottom',
            itemMargin: {
                horizontal: 2,
                vertical: 5
            }
        },
        theme: {
            mode: 'dark',
            palette: 'palette1',
        },
        grid: {
            borderColor: '#0000',
            strokeDashArray: 7,
        },
        xaxis: {
            lines: {
                show: false,
            }
        },
        yaxis: {
            lines: {
                show: true,
            }
        },
        labels: labels,
        series: values,
        responsive: [{
            breakpoint: 480,
            options: {
                chart: {
                    width: 350
                },
                legend: {
                    show: false
                }
            }
        }]
    }

    return(
        <React.Fragment>
            <Grid container spacing={theme.spacing(1)} alignContent={'space-between'} alignItems={'flex-start'}>
                <Grid item xs={12} lg={12}>
                    { currentCourseWeighting.length > 0 &&
                    <Grow
                        in={currentCourseWeighting.length > 0}
                        style={{ transformOrigin: '0 0 0' }}
                        timeout={1000}
                    >
                    <Paper elevation={4}  className={classes.paper}>


                        <Grid container spacing={theme.spacing(0)} >
                            <Grid item xs={12} sm={12} md={12}>
                                <Typography variant={'h5'}>Content Standards or Topic Weighting</Typography>
                            </Grid>
                            <Grid item xs={12} sm={6} md={4}>
                            <Table size={'small'}>
                                <TableHead>
                                    <TableRow>
                                        <TableCell colSpan={15} >Topic Name</TableCell>
                                        <TableCell align='center' colSpan={2} >Weighting</TableCell>
                                    </TableRow>
                                </TableHead>
                                <TableBody>
                                { currentCourseWeighting.map( (weighting, index) => {
                                    return (
                                   <TableRow key={`weight-${weighting.id}`}>
                                           <TableCell colSpan={15}>{weighting.name}</TableCell>
                                           <TableCell  align='center' ><FormControl className={''} fullWidth={true} >
                                               <TextField
                                                   name={'weight-field'}
                                                   value={weighting.weight}
                                                   onChange={(event) => {handleChangeWeight(event, weighting) }}
                                                   onBlur={(event) => {handleBlurWeight(event, weighting)}}
                                               >
                                               </TextField>
                                           </FormControl></TableCell>
                                       </TableRow>
                                   )
                            })}
                                </TableBody>
                            </Table>
                            </Grid>

                            <Grid item xs={12} sm={6} md={8}>
                                <ReactApexChart options={chartOptions} series={values} type="pie" height="350"/>

                            </Grid>
                            <Grid item xs={12} sm={6} md={8}>
                                <Button color={'secondary'} onClick={handleSaveWeights} disabled={saveIsDisabled()}>Save</Button>
                            </Grid>

                        </Grid>

                    </Paper>
                    </Grow>
                        }
                </Grid>
            </Grid>

        </React.Fragment>
        )

}
