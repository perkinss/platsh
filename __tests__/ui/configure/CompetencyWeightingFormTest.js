import React from 'react'
import TopicWeightingForm from '../../../app/javascript/components/configure/TopicWeightingForm'
import { createRender } from '@material-ui/core/test-utils';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import expect from 'expect'
import theme from "../../../app/javascript/styles/theme";
import {ThemeProvider } from '@material-ui/styles'
import {ConfigurationContext} from "../../../app/javascript/context/ConfigurationContext";
import CompetencyWeightingForm from "../../../app/javascript/components/configure/CompetencyWeightingForm";

Enzyme.configure({ adapter: new Adapter() });

describe('<CompetencyWeightingForm />', () => {
    let render;
    let classes = {}
    let state = {
        sections: [
            {
                courses: [
                    {id: 5, title: "Course 1 of course"},
                    {id: 2, title: "Course 2 of course"}
                ]
            },
            {
                courses: [
                    {id: 6, title: "Course 5 of course"},
                    {id: 2, title: "Course 2 of course"}
                ]
            }
        ]
    }

    beforeAll(() => {
        render = createRender();
    })

    it('should match the snapshot', () => {
        const wrapper = render(
            <ThemeProvider theme={theme}>
                <ConfigurationContext.Provider value={{dispatch: () => {}, state: state }} >
                    <CompetencyWeightingForm classes={classes} />
                </ConfigurationContext.Provider>
            </ThemeProvider>);
        expect(wrapper).not.toBeNull()
        expect(wrapper).toMatchSnapshot()
    });

})