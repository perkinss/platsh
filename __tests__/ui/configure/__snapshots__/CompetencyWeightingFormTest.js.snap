// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CompetencyWeightingForm /> should match the snapshot 1`] = `
initialize {
  "0": Node {
    "attribs": Object {
      "class": "MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-8 MuiGrid-align-items-xs-flex-start MuiGrid-align-content-xs-space-between",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "class": "MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-lg-12",
        },
        "children": Array [],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
